cmake/modules/hunter_boost_component_b2_args.cmake

```diff
@@ -35,46 +39,79 @@ function(hunter_boost_component_b2_args compName boostCmakeArgs outList)
         list(GET optValue 1 v)
 
         if(NOT is_nocompress)
-          string(COMPARE EQUAL ${n} "NO_COMPRESSION" is_nocompress)
-          if(is_nocompress)
-            set(is_nocompress ${v})
+          string(COMPARE EQUAL ${n} "NO_COMPRESSION" specified)
+          if(specified AND v)
+            set(is_nocompress "1")
           endif()
         endif()
-        if(NOT is_nobzip)
-          string(COMPARE EQUAL ${n} "NO_BZIP2" is_nobzip)
-          if(is_nobzip)
-            set(is_nobzip ${v})
+        if(NOT is_nobzip2)
+          string(COMPARE EQUAL ${n} "NO_BZIP2" specified)
+          if(specified AND v)
+            set(is_nobzip2 "1")
           endif()
         endif()
         if(NOT is_nozlib)
-          string(COMPARE EQUAL ${n} "NO_ZLIB" is_nozlib)
-          if(is_nozlib)
-            set(is_nozlib ${v})
+          string(COMPARE EQUAL ${n} "NO_ZLIB" specified)
+          if(specified AND v)
+            set(is_nozlib "1")
           endif()
         endif()
 
-        list(APPEND myList "-s" "${compNvPair}")
       endif()
     endforeach()
+
+    list(APPEND myList "-s" "NO_COMPRESSION=${is_nocompress}")
+
+    if(NOT is_nozlib)
+      list(APPEND myList "-s" "NO_ZLIB=0")
+    else()
+      list(APPEND myList "-s" "NO_ZLIB=1")
+    endif()
+
+    if(NOT is_nobzip2)
+      list(APPEND myList "-s" "NO_BZIP2=0")
+    else()
+      list(APPEND myList "-s" "NO_BZIP2=1")
+    endif()
+
     if(NOT is_nocompress)
+      if(CMAKE_BUILD_TYPE STREQUAL "Debug")
```

Comment:
This is not correct in general (see [so answer](http://stackoverflow.com/a/24470998/2288008)) and will not work in our case since this function called from project with `ExternalProject_Add` command (i.e. super-project build both release and debug variant).

Also this bug need to be solved first: https://github.com/ruslo/hunter/issues/146